model_evaluation/jigsaw_evaluation_pipeline.ipynb

```diff
@@ -293,6 +302,109 @@
+    "    'male': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'female': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'transgender': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'other_gender': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'heterosexual': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'homosexual_gay_or_lesbian': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'bisexual': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'other_sexual_orientation': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'christian': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'jewish': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'muslim': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'hindu': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'buddhist': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'atheist': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'other_religion': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'black': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'white': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'asian': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'latino': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'other_race_or_ethnicity': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'physical_disability': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'intellectual_or_learning_disability': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'psychiatric_or_mental_illness': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "    'other_disability': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.),\n",
+    "}\n",
+    "\n",
+    "identity_terms_civil = ['male', 'female', 'transgender', 'other_gender', 'heterosexual', \n",
+    "                        'homosexual_gay_or_lesbian', 'bisexual', 'other_sexual_orientation',\n",
+    "                        'christian', 'jewish', 'muslim', 'hindu', 'buddhist', 'atheist',\n",
+    "                        'other_religion', 'black', 'white', 'asian', 'latino', 'other_race_or_ethnicity',\n",
+    "                        'physical_disability', 'intellectual_or_learning_disability',\n",
+    "                        'psychiatric_or_mental_illness', 'other_disability'\n",
+    "                       ]\n",
+    "\n",
+    "def filter_fn_civil(example, background_filter_keep_rate=0.1):\n",
+    "    if example ['male'] < 0.:\n",
```

Comment:
How can example['male'] be < 0? Should this be <= 0 ?

Reply:
If you look at the way I parsed the tf-records:
'male': tf.FixedLenFeature([], dtype=tf.float32, default_value=-1.)

Because male is an optional field, I need to provide a default_value. When male is not in the record, it will take the default value. I have not heard of any way to say "Parse only existing fields".

So taking ('male' <0) means we remove the lines where there was no identity information. From what I have seen, information is provided for either all entities or none. So (male>=0) means that we have information for all entities.

Does that make sense?

Reply:
Yes, thanks for clarifying!